megamek/src/megamek/client/ui/swing/PlanetaryConditionsDialog.java

```diff
@@ -148,6 +149,23 @@ public void windowClosing(WindowEvent e) {
             }
         });
 
+        choWeather.addItemListener(e -> {
```

Comment:
In Swing an `ItemListener` fires twice when the selection changes: once for deselecting the previous item and again for selecting the new item. An `ActionListener` only fires once when the selection changes, and is preferred except in the rare cases when you want to know which item was previously selected.